src/module.c

```diff
@@ -12272,35 +12533,13 @@ NULL
         }
 
     } else if (!strcasecmp(subcmd,"unload") && c->argc == 3) {
-        if (moduleUnload(c->argv[2]->ptr) == C_OK)
+        const char *errmsg = NULL;
+        if (moduleUnload(c->argv[2]->ptr, &errmsg) == C_OK)
             addReply(c,shared.ok);
         else {
-            char *errmsg;
-            switch(errno) {
-            case ENOENT:
-                errmsg = "no such module with that name";
-                break;
-            case EBUSY:
-                errmsg = "the module exports one or more module-side data "
-                         "types, can't unload";
-                break;
-            case EPERM:
-                errmsg = "the module exports APIs used by other modules. "
-                         "Please unload them first and try again";
-                break;
-            case EAGAIN:
-                errmsg = "the module has blocked clients. "
-                         "Please wait them unblocked and try again";
-                break;
-            case EINPROGRESS:
-                errmsg = "the module holds timer that is not fired. "
-                         "Please stop the timer or wait until it fires.";
-                break;
-            default:
-                errmsg = "operation not possible.";
-                break;
-            }
-            addReplyErrorFormat(c,"Error unloading module: %s",errmsg);
+            if (errmsg == NULL) errmsg = "operation not possible.";
+            addReplyErrorFormat(c, "Error unloading module: %s", errmsg);
+            serverLog(LL_WARNING, "Error unloading module %s: %s", (sds) c->argv[2]->ptr, errmsg);
```

Comment:
```suggestion
            serverLog(LL_WARNING, "Error unloading module %s: %s", c->argv[2]->ptr, errmsg);
```

Reply:
```
module.c:12542:68: warning: format specifies type 'char *' but the argument has type 'void *' [-Wformat]
            serverLog(LL_WARNING, "Error unloading module %s: %s", c->argv[2]->ptr, errmsg);
                                                          ~~       ^~~~~~~~~~~~~~~
./server.h:3646:27: note: expanded from macro 'serverLog'
        _serverLog(level, __VA_ARGS__);\
```

I attempted this change and saw that it raises this warning